adafruit_requests.py

```diff
@@ -561,40 +578,48 @@ def _send_request(
         data: Any,
         json: Any,
     ):
-        # pylint: disable=too-many-arguments
-        self._send(socket, bytes(method, "utf-8"))
-        self._send(socket, b" /")
-        self._send(socket, bytes(path, "utf-8"))
-        self._send(socket, b" HTTP/1.1\r\n")
-        if "Host" not in headers:
-            self._send(socket, b"Host: ")
-            self._send(socket, bytes(host, "utf-8"))
-            self._send(socket, b"\r\n")
-        if "User-Agent" not in headers:
-            self._send(socket, b"User-Agent: Adafruit CircuitPython\r\n")
-        # Iterate over keys to avoid tuple alloc
-        for k in headers:
-            self._send(socket, k.encode())
-            self._send(socket, b": ")
-            self._send(socket, headers[k].encode())
-            self._send(socket, b"\r\n")
+        # Convert data
```

Comment:
Updated method to
1. modify data and determine possible `Content-Type` header
2. send init bytes
3. send `host` and `user agent` headers (when not supplied)
4. send `content type` and `content length` headers (when not supplied)
5. send supplied headers 
6. send data